juju/api.go

```diff
@@ -58,7 +58,7 @@ func NewAPIConnection(args NewAPIConnectionParams) (_ api.Connection, err error)
 		return nil, errors.Annotatef(err, "cannot work out how to connect")
 	}
 	if len(apiInfo.Addrs) == 0 {
-		return nil, errors.New("no API addresses")
+		return nil, errors.New("no controller API addresses; is bootstrap still in progress?")
```

Comment:
Yeah, it would be better if the api layer avoid user facing messaging - that's the responsibility of the CLI. So returning a specific error type would be better.